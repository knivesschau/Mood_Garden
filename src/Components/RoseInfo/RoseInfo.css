/* Mobile styles first, 320px */

.Collapsed_Roses, .Rose_MiniInfo {
    transition-property: background; 
    transition-duration: .5s;
    transition-timing-function: ease-in;
    border: 1px transparent white;
    border-radius: 20em;
}

.Collapsed_Roses:hover {
    background-color: #A88087;
    cursor: pointer;
    box-shadow: inset -11px -5px 5px 0px #6E4D65;
}

.Collapsed_Roses:active {
    background-color: #7B424C;
    box-shadow: inset -11px -5px 5px 0px #6E4D65;
}

.Rose_Image_Garden {
    position: relative;
    right: 7.75em;
    bottom: .5em;
}

#rose-info {
    font-weight: bold;
    font-size: 1.25em;
    word-spacing: -.10rem;
    margin-top: -2.5em;
    margin-left: 2em; 
    margin-right: 2em;
    padding: .25em;
}

#rose-info:hover {
    color: white;
}

@media only screen and (min-device-width: 375px) {
    .Rose_Image_Garden {
        right: 9em;
    }

    #rose-info {
        margin-top: -3em;
        padding: .50em;
    }
}

@media only screen and (min-device-width: 414px) {
    .Rose_Image_Garden {
        right: 10em;
    }

    #rose-info {
        font-size: 1.5em;
        margin-left: 2.5em;
        margin-right: 2.5em;
    }
}

@media only screen and (min-device-width: 768px) {
    .Rose_Image_Garden {
        right: 15em;
    }

    #rose-info {
        padding: 1em;
    }
}

@media only screen and (min-device-width: 1000px) {
    .Rose_Image_Garden {
        right: 20em;
        bottom: .50em;
    }

    #rose-info {
        font-size: 1.75em;
    }
}

@media only screen and (min-device-width: 1200px) {
    .Rose_Image_Garden {
        right: 17.5em;
    }
}

@media only screen and (min-device-width: 1300px) {
    .Rose_Image_Garden {
        right: 20em;
    }

    #rose-info {
        font-size: 2em;
    }
}

@media only screen and (min-device-width: 1500px) {
    .Rose_Image_Garden {
        right: 21em;
    }
}

@media only screen and (min-device-width: 1700px) {
    .Rose_Image_Garden {
        right: 22em;
    }

    #rose-info {
        font-size: 2.25em;
    }
}

@media only screen and (min-device-width: 2000px) {
    .Rose_Image_Garden {
        right: 27em;
    }
}