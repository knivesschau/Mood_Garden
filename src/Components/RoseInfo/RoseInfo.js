import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import moodGardenContext from '../../moodGardenContext';
import './RoseInfo.css';

export default class RoseInfo extends Component {

    static contextType = moodGardenContext;

    render() {
        const {id, color, entry_date} = this.props;

        return (
            <section className="ViewRose">
            
            <div className="Collapsed_Roses">
                <Link to={`/roses/${id}`}>
                    <h2>{color} Rose Planted on {new Date(entry_date).toLocaleDateString()}</h2>
                </Link>
            </div>         

        </section>
        );
    }
}