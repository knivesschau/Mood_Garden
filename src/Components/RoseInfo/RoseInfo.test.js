import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import RoseInfo from './RoseInfo';

describe ('RoseInfo Component', () => {
    it ('renders the complete RoseInfo Component', () => {
        const wrapper = shallow(<RoseInfo/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})