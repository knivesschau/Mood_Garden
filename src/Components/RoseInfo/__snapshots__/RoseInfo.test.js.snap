// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoseInfo Component renders the complete RoseInfo Component 1`] = `
<section
  className="ViewRose"
>
  <div
    className="Collapsed_Roses"
  >
    <Link
      to="/roses/undefined"
    >
      <h2>
         Rose Planted on 
        Invalid Date
      </h2>
    </Link>
  </div>
</section>
`;
